src/integration-test/java/com/scalar/db/storage/cosmos/CosmosIntegrationTest.java

```diff
@@ -339,6 +402,56 @@ public void scan_ScanWithLimitGiven_ShouldReturnGivenNumberOfResults() throws Ex
         .isEqualTo(Optional.of(new IntValue(COL_NAME4, 2)));
   }
 
+  @Test
+  public void scan_ScanGivenForIndexedColumn_ShouldScan() throws ExecutionException {
+    // Arrange
+    populateRecords();
+    int c3 = 3;
+    Scan scan = new Scan(new Key(new IntValue(COL_NAME3, c3)));
+
+    // Act
+    List<Result> actual = storage.scan(scan).all();
+
+    // Assert
+    assertThat(actual.size()).isEqualTo(3); // (1,2), (2,1), (3,0)
+    assertThat(actual.get(0).getValue(COL_NAME1))
+        .isEqualTo(Optional.of(new IntValue(COL_NAME1, 1)));
+    assertThat(actual.get(0).getValue(COL_NAME4))
+        .isEqualTo(Optional.of(new IntValue(COL_NAME4, 2)));
+    assertThat(actual.get(1).getValue(COL_NAME1))
+        .isEqualTo(Optional.of(new IntValue(COL_NAME1, 2)));
+    assertThat(actual.get(1).getValue(COL_NAME4))
+        .isEqualTo(Optional.of(new IntValue(COL_NAME4, 1)));
+    assertThat(actual.get(2).getValue(COL_NAME1))
+        .isEqualTo(Optional.of(new IntValue(COL_NAME1, 3)));
+    assertThat(actual.get(2).getValue(COL_NAME4))
+        .isEqualTo(Optional.of(new IntValue(COL_NAME4, 0)));
+  }
+
+  @Test
+  public void scan_ScanGivenForIndexedColumnWithOrdering_ShouldThrowIllegalArgumentException()
+      throws ExecutionException {
+    // Arrange
+    int c3 = 0;
+    Scan scan =
+        new Scan(new Key(new IntValue(COL_NAME3, c3)))
+            .withOrdering(new Scan.Ordering(COL_NAME4, Scan.Ordering.Order.ASC));
+
+    // Act
+    assertThatThrownBy(() -> storage.scan(scan)).isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void scan_ScanGivenForNonIndexedColumn_ShouldThrow() {
```

Comment:
```suggestion
  public void scan_ScanGivenForNonIndexedColumn_ShouldThrowIllegalArgumentException() {
```
Sorry, I forgot to add the exception.